src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,98 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
-
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    
+    private Node<K, V>[] table;
+    private int threshold;
+    private int size;
+    
+    @SuppressWarnings("unchecked")
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
+    
+    private static class Node<K, V> {
+        private final int hash;
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+        
+        Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+    
     @Override
     public void put(K key, V value) {
-
+        int hash = hash(key);
+        Node<K, V> currentNode = table[hash];
+        Node<K, V> newNode = new Node<>(hash, key, value, null);
+        if (currentNode == null) {
+            table[hash] = newNode;
+        } else {
+            while (true) {
+                if (key == currentNode.key || (key != null && key.equals(currentNode.key))) {
+                    currentNode.value = value;
+                    return;
+                }
+                if (currentNode.next == null) {
+                    currentNode.next = newNode;
+                    break;
```

Comment:
We can leave the while loop as long as we have reached the end of the list at this index, then append a new `Node` to the end of the list